contracts/Zauction.sol

```diff
@@ -76,27 +92,109 @@ contract ZAuction is Initializable, OwnableUpgradeable {
     uint256 bidNonce,
     address bidder,
     uint256 bid,
-    uint256 tokenId,
+    uint256 domainTokenId,
     uint256 minbid,
     uint256 startBlock,
     uint256 expireBlock
+  ) external {
+    require(startBlock <= block.number, "zAuction: auction hasn't started");
+    require(expireBlock > block.number, "zAuction: auction expired");
+    require(minbid <= bid, "zAuction: cannot accept bid below min");
+    require(bidder != msg.sender, "zAuction: cannot sell to self");
+    require(msg.sender == hub.ownerOf(domainTokenId), "Only Owner");
+
+    IRegistrar domainRegistrar = hub.getRegistrarForDomain(domainTokenId);
+
+    // Don't include a token address when recreating bid data for legacy bids
```

Comment:
You are including a token address, it's just `address(0)`
Leave a comment about how address(0) makes it create a legacy bid